backends/edk2-compat/edk2-svc-read.c

```diff
@@ -16,458 +16,435 @@
-		// puts sig data in cert
-		cert_size = get_esl_cert(c + offset, sigList, (char **)&cert); 
-		if (cert_size <= 0) {
-			prlog(PR_ERR, "\tERROR: Signature Size was too small, no data \n");
-			break;
-		}
-		if (key && !strcmp(key, "dbx")) {
-			printf("\tHash: ");
-			printHex(cert, cert_size);
-		}
-		else {
-			x509 = malloc(sizeof(*x509));
-			if (!x509) {
-				prlog(PR_ERR, "ERROR: failed to allocate memory\n");
-				return ALLOC_FAIL;
-			}
-			rc = parseX509(x509, cert, (size_t) cert_size);
-			if (rc)
-				break;
-			rc = printCertInfo(x509);
-			if (rc)
-				break;
-			free(cert);
-			cert = NULL;
-			mbedtls_x509_crt_free(x509);
-			free(x509);
-			x509 = NULL;
-		}
+// 	return rc;
+// }
+
+// /**
+//  *Does the appropriate read command depending on hrFlag on the file 
+//  *@param file , the path to the file 
+//  *@param hrFlag, 1 for human readable 0 for raw data
+//  *@return SUCCESS or error number
+//  */
+// static int readFileFromPath(const char *file, int hrFlag)
+// {
+// 	int rc;
+// 	size_t size = 0;
+// 	char *c = NULL;
+// 	c = getDataFromFile(file, &size);
+// 	if (!c) {
+// 		return INVALID_FILE;
+// 	}
+// 	if (hrFlag) {
+// 		rc = printReadable(c, size, NULL);
+// 		if(rc)
+// 			prlog(PR_WARNING,"ERROR: Could not parse file\n");
+// 		else
+// 			rc = SUCCESS; 		
+// 	}
+// 	else {
+// 		printRaw(c, size);
+// 		rc = SUCCESS;
+// 	}
+// 	free(c);
+
+// 	return rc;
+// }
+
+// /**
+//  *gets the secvar struct from a file
+//  *@param var , returned secvar
+//  *@param name , secure variable name {db,dbx,KEK,PK}
+//  *@param fullPath, file and path <path>/<varname>/data
+//  *NOTE: THIS IS ALLOCATING DATA AND var STILL NEEDS TO BE DEALLOCATED
+//  */
+// int getSecVar(struct secvar **var, const char* name, const char *fullPath){
+// 	int rc, fptr;
+// 	size_t size;
+// 	ssize_t read_size;
+// 	char *sizePath = NULL, *c = NULL;
+// 	struct stat fileInfo;
+// 	rc = isFile(fullPath);
+// 	if (rc) {
+// 		return rc;
+// 	}
+// 	sizePath = malloc( strlen(fullPath) + 1);
+// 	if (!sizePath) {
+// 		prlog(PR_ERR, "ERROR: failed to allocate memory\n");
+// 		return ALLOC_FAIL;
+// 	}
+// 	// since we are reading from a secvar, it can be assumed it has a <var>/size file for more accurate size
+// 	// fullPath currently holds <path>/<var>/data we are going to take off data and add size to get the desired file
+// 	strcpy(sizePath, fullPath);
+// 	//add null terminator so strncat works
+// 	sizePath[strlen(sizePath) - strlen("data")] = '\0';
+// 	strncat(sizePath, "size", strlen("size") + 1); 
+// 	rc = getSizeFromSizeFile(&size, sizePath);
+// 	if (rc < 0) {
+// 		prlog(PR_WARNING, "ERROR: Could not get size of variable, TIP: does %s exist?\n", sizePath);
+// 		rc = INVALID_FILE;
+// 		free(sizePath);
+// 		return rc;
+// 	}
+// 	free(sizePath);
+
+// 	if (size == 0) {
+// 		prlog(PR_WARNING, "Secure Variable has size of zero, (specified by size file)\n");
+// 		/*rc = INVALID_FILE;
+// 		return rc;*/
+// 	}
+
```

Comment:
I'm assuming something broke when you started re-implementing the crypto, hence this giiiiant diff of half the file commented out. It's probably not worth the effort to fix now, but in the future try to avoid commits like this that comment out large swaths of code temporarily.

Instead, try to make breaking changes either incrementally, or write all the new code elsewhere and then add a commit that switches over to the new code. For example, you could have added the crypto files as a commit, but not build/use/reference them until a later commit that rewrites the commands with the new crypto functions.

Reply:
You can also use `#if 0` if you're trying to comment out huge chunks. But as Eric says, avoiding it entirely is better. Something he doesn't mention is bisect-ability - see https://wiki.samba.org/index.php/Creating_a_Samba_patch_series#Bisectability - which is something that IMO is worth preserving at least in general. In your case, I think it should be possible  for the entire program to compile with mbedtls the whole way through...